Source/src/core/Scene.h

```diff
@@ -116,8 +115,46 @@ namespace Hachiko
             name = new_name;
         }
 
+        float GetAmbientIntensity() const
+        {
+            return ambient_light_intensity;
+        }
+
+        const float4& GetAmbientColor()
+        {
+            return ambient_light_color;
+        }
+
+        bool IsFogEnabled() const
+        {
+            return fog_enabled;
+        }
+
+        const float3& GetFogColor() const
+        {
+            return fog_color;
+        }
+
+        const float* GetFogGlobalDensity() const
+        {
+            return &fog_global_density;
+        }
+
+        const float* GetFogHeightFalloff() const 
+        {
+            return &fog_height_falloff;
+        }
```

Comment:
I would encapsulate fog related stuff inside a small struct/class that is called FogConfig and have scene own one to not convolute the scene class any further maybe. Then the API would be `Scene::fog_config.IsFogEnabled` for example.

Reply:
Done <3